test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateE2ETests.cs

```diff
@@ -319,6 +353,108 @@ public void BulkUpdateTwoTopLevelObjects_WithTheSameNestedObject_DeserializesSuc
             Assert.Equal(carTargetEntity, car2TargetEntity);
         }
 
+        [Fact]
+        public async Task BulkUpdateAsyncTwoTopLevelObjects_WithTheSameNestedObject_DeserializesSuccessfully()
+        {
+            var expectedResponse = "{\"@context\":\"http://localhost:8000/$metadata#Persons/$delta\",\"value\":[{\"ID\":100,\"Name\":\"Bing\",\"Cars@delta\":[{\"@id\":\"http://localhost:8000/Cars(1001)\"}]},{\"ID\":200,\"Name\":\"Edge\",\"Cars@delta\":[{\"@id\":\"http://localhost:8000/Cars(1001)\"}]}]}";
+
+            SetupContextWithRequestPipelineForSaving(
+                this.context,
+                expectedResponse);
+
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var person2 = new Person
+            {
+                ID = 200,
+                Name = "Edge"
+            };
+
+            var car = new Car
+            {
+                ID = 1002,
+                Name = "CarA"
+            };
+
+            this.context.AttachTo("Persons", person);
+            this.context.AttachTo("Persons", person2);
+            this.context.AttachTo("Cars", car);
+
+            this.context.AddLink(person, "Cars", car);
+            this.context.AddLink(person2, "Cars", car);
+
+            DataServiceResponse response = await this.context.BulkUpdateAsync(person, person2);
+
+            var personOperationResponse = response.First() as ChangeOperationResponse;

```

Comment:
can we wrapper those lines into a verifier method? It seems it duplicates.

Reply:
We test for different things here. Wrapping these lines might not be of much help.